test/e2e/ha_master.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"os/exec"
+	"path"
+	"regexp"
+	"strconv"
+)
+
+func addMasterReplica() error {
+	framework.Logf(fmt.Sprintf("Adding a new master replica:"))
+	v, _, err := framework.RunCmd(path.Join(framework.TestContext.RepoRoot, "hack/e2e-internal/e2e-add-master.sh"))
+	if err != nil {
+		return err
+	}
+	framework.Logf("%q", v)
+	return nil
+}
+
+func removeMasterReplica() error {
+	framework.Logf(fmt.Sprintf("Removing an existing master replica:"))
+	v, _, err := framework.RunCmd(path.Join(framework.TestContext.RepoRoot, "hack/e2e-internal/e2e-remove-master.sh"))
+	if err != nil {
+		return err
+	}
+	framework.Logf("%q", v)
+	return nil
+}
+
+func verifyRCs(c *client.Client, ns string, names []string) {
+	for _, name := range names {
+		framework.ExpectNoError(framework.VerifyPods(c, ns, name, true, 1))
+	}
+}
+
+func createNewRC(c *client.Client, ns string, name string) {
+	_, err := newRCByName(c, ns, name, 1)
+	framework.ExpectNoError(err)
+}
+
+func verifyNumberOfMasterReplicas(expected int) {
+	output, err := exec.Command("gcloud", "compute", "instances", "list",
+		"--project="+framework.TestContext.CloudConfig.ProjectID,
+		"--zone="+framework.TestContext.CloudConfig.Zone,
+		"--regexp="+framework.TestContext.CloudConfig.MasterName + "(-...)?").CombinedOutput()
+	framework.ExpectNoError(err)
+	re := regexp.MustCompile(".*RUNNING")
```

Comment:
Still doable:

```
$ gcloud compute instances list --regexp=".*master.*" --filter=status=RUNNING --format=[no-heading]
kubernetes-master  us-central1-b  n1-standard-1    10.128.0.8  146.148.92.107  RUNNING
```

See [here](https://cloud.google.com/sdk/gcloud/reference/topic/formats) for more formats.